rocAL/include/loaders/image/numpy_loader.h

```diff
@@ -0,0 +1,94 @@
+    ~NumpyLoader() override;
+    LoaderModuleStatus load_next() override;
+    void initialize(ReaderConfig reader_cfg, DecoderConfig decoder_cfg, RocalMemType mem_type, unsigned batch_size, bool keep_orig_size = false) override;
+    void set_output(Tensor* output_image) override;
+    void set_random_bbox_data_reader(std::shared_ptr<RandomBBoxCrop_MetaDataReader> randombboxcrop_meta_data_reader) override;
+    size_t remaining_count() override;  // returns number of remaining items to be loaded
+    void reset() override;              // Resets the loader to load from the beginning of the media
+    Timing timing() override;
+    void start_loading() override;
+    LoaderModuleStatus set_cpu_affinity(cpu_set_t cpu_mask);
+    LoaderModuleStatus set_cpu_sched_policy(struct sched_param sched_policy);
+    void set_gpu_device_id(int device_id);
+    std::vector<std::string> get_id() override;
+    DecodedDataInfo get_decode_data_info() override;
+    CropImageInfo get_crop_image_info() override;
+    void set_prefetch_queue_depth(size_t prefetch_queue_depth) override;
+    void shut_down() override;
+    void feed_external_input(const std::vector<std::string>& input_images_names, const std::vector<unsigned char*>& input_buffer,
+                             const std::vector<ROIxywh>& roi_xywh, unsigned int max_width, unsigned int max_height, unsigned int channels, ExternalSourceFileMode mode, bool eos) override {};
+    size_t last_batch_padded_size() override;
+
+   private:
+    bool is_out_of_data();
+    void de_init();
+    void stop_internal_thread();
+    LoaderModuleStatus update_output_image();
+    LoaderModuleStatus load_routine();
+    std::shared_ptr<Reader> _reader;
+    std::shared_ptr<RandomBBoxCrop_MetaDataReader> _randombboxcrop_meta_data_reader = nullptr;
+    Tensor* _output_tensor;
+    std::vector<std::string> _output_names;  //!< image name/ids that are stores in the _output_image
+    size_t _output_mem_size;
+    MetaDataBatch* _meta_data = nullptr;  //!< The output of the meta_data_graph,
+    std::vector<std::vector<float>> _bbox_coords;
+    bool _internal_thread_running;
+    size_t _batch_size;
+    size_t _image_size;
+    std::thread _load_thread;
+    RocalMemType _mem_type;
+    DecodedDataInfo _decoded_data_info;
+    CropImageInfo _crop_image_info;
+    DecodedDataInfo _output_decoded_data_info;
+    CropImageInfo _output_cropped_img_info;
+    CircularBuffer _circ_buff;
+    TimingDbg _file_load_time, _swap_handle_time;
+    bool _is_initialized;
+    bool _stopped = false;
+    bool _loop;                     //<! If true the reader will wrap around at the end of the media (files/images/...) and wouldn't stop
+    size_t _prefetch_queue_depth;   // Used for circular buffer's internal buffer
+    size_t _image_counter = 0;      //!< How many images have been loaded already
+    size_t _remaining_image_count;  //!< How many images are there yet to be loaded
+    bool _decoder_keep_original = false;
```

Comment:
This seems to be unused in numpy_loader.cpp. Please check & remove unused variables